util/topgen/merge.py

```diff
@@ -885,20 +889,36 @@ def create_alert_lpgs(top, name_to_block: Dict[str, IpBlock]):
 
         # if clock group is "unique", add some uniquification to the tag
         lpg_name = f"{module['name']}_{lpg_name}" if unique_cg else lpg_name
-        if lpg_name not in lpg_dict:
-            lpg_dict.update({lpg_name: num_lpg})
-            # since the alert handler can tolerate timing delays on LPG
-            # indication signals, we can just use the clock / reset signals
-            # of the first block that belongs to a new unique LPG.
-            clock = module['clock_connections'][block_clock.clock]
-            top['alert_lpgs'].append({
+
+        def append_to_lpg_dict(lpg_dict):
+            lpg_dict.append({
                 'name': lpg_name,
                 'clock_group': None if unmanaged_clock else clock_group,
                 'clock_connection': clock,
                 'unmanaged_clock': unmanaged_clock,
                 'reset_connection': primary_reset
             })
-            num_lpg += 1
+
+        if 'outgoing_alert' in module:
+            alert_group = module['outgoing_alert']
+            if lpg_name not in outgoing_lpg_dict[alert_group]:
+                outgoing_lpg_dict[alert_group].update({lpg_name: outgoing_num_lpg[alert_group]})
+
+                # since the alert handler can tolerate timing delays on LPG
+                # indication signals, we can just use the clock / reset signals
+                # of the first block that belongs to a new unique LPG.
+                clock = module['clock_connections'][block_clock.clock]
+                append_to_lpg_dict(top['outgoing_alert_lpgs'][alert_group])
+                outgoing_num_lpg[alert_group] += 1
+        else:
+            if lpg_name not in lpg_dict:
+                lpg_dict.update({lpg_name: num_lpg})
+                # since the alert handler can tolerate timing delays on LPG
+                # indication signals, we can just use the clock / reset signals
+                # of the first block that belongs to a new unique LPG.
+                clock = module['clock_connections'][block_clock.clock]
+                append_to_lpg_dict(top['alert_lpgs'])
+                num_lpg += 1
```

Comment:
Can't we get rid of `num_lpg` and just use `len(lpg_dict)`?

Reply:
No I don't think that would work. We need a mapping of LPG name to the index in the vector to be able to later associate the alerts to that.

Reply:
I think I agree that we should keep the dictionary, but I'm reasonably sure that `num_lpg` is always equal to `len(lpg_dict)`, so I'm proposing that we get rid of the `num_lpg` variable.

Reply:
Ahhh, sorry. I misread that. Got it now. Yeah, as we construct the dict, we can use its length to get the mapping index.

Reply:
Looks good, thanks!